bigstore/bigstore.py

```diff
@@ -322,9 +324,18 @@ def pull():
 
                         break
 
-    sys.stderr.write("pushing bigstore metadata...")
-    g().push("origin", "refs/notes/bigstore")
-    sys.stderr.write("done\n")
+    if g().diff('refs/notes/bigstore', 'refs/notes/bigstore-remote'):
```

Comment:
It is unclear to me why `git bigstore pull` should *ever* call `git push`.  Having the `pull` command also push metadata might be confusing to users and has the side effect of requiring write access in order to read from a repository.

I am tempted to just delete this block of code... can you explain the purpose of it to me?